internal/controller/instaslice_controller.go

```diff
@@ -523,9 +526,25 @@ func (r *InstasliceReconciler) setupWithManager(mgr ctrl.Manager) error {
 	if err != nil {
 		return err
 	}
+	// pod filtering
+	// only watches Pods that:
+	// have the mutation annotation, or
+	// have meaningful Change
+	instaslicePredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return hasInstasliceMutation(e.Object)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return hasInstasliceMutation(e.ObjectNew) || !isEqual(e.ObjectOld, e.ObjectNew)
```

Comment:
I think this has to be the other way
* NewObj should have the required annotation
* And there is a change in the `spec` or the `status` sub structs.
Otherwise, this would always be triggered with logical `OR` if the new obj just has the annotation and no meaningful change
```suggestion
			return hasInstasliceMutation(e.ObjectNew) && !isEqual(e.ObjectOld, e.ObjectNew)
```

Reply:
good catch